### PR TITLE
migrate build to tsup

### PR DESCRIPTION
**Pull request by kyscott18**
Migrate from tsc to tsup for bulid. Remove multiple packages previously supporting our build config. The end to end tests were also cleaned up a big because of the new tsconfig. The build step was separated from the actions setup workflow as we aim to be able to test before building.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,12 +25,32 @@ jobs:
...
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+
+      - name: Setup
+        uses: ./.github/actions/setup
+
+      - name: Build
+        run: pnpm build
+
```
Comment (0xOlias) [inline .github/workflows/main.yml:53]: What's the purpose of this?

File: packages/core/src/_test/art-gobblers/app/src/index.ts [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+// @ts-ignore
 ponder.on("setup", async ({ context }) => {
   const { SetupEntity } = context.db;
 
   await SetupEntity.upsert({
     id: "setup_id",
-    create: {},
-    update: {},
+    create: undefined,
+    update: undefined,
```
Comment (0xOlias) [inline packages/core/src/_test/art-gobblers/app/src/index.ts:18]: I think you can just omit these lines?

File: packages/core/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
+    "allowSyntheticDefaultImports": true,
+    "resolveJsonModule": true,
+
+    // Language and environment
+    "moduleResolution": "NodeNext",
+    "module": "NodeNext",
+    "target": "ESNext",
+    "lib": [
+      "ES2022", // By using ES2022 we get access to the `.cause` property on `Error` instances.
+      "DOM" // We are adding `DOM` here to get the `fetch`, etc. types. This should be removed once these types are available via DefinitelyTyped.
```
Comment (0xOlias) [inline packages/core/tsconfig.json:36]: Wonder if we need this

File: packages/create-ponder/tsup.config.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { defineConfig } from "tsup";
 import { dependencies } from "./package.json";
 
 export default defineConfig({
+  name: "create-ponder",
```
Comment (0xOlias) [inline packages/create-ponder/tsup.config.ts:10]: What does the name field do?

File: .github/workflows/main.yml (hunk 1 above)
Comment (kyscott18) [reply .github/workflows/main.yml:53]: It is a test to see if the packages are able to be built successfully

File: packages/create-ponder/tsup.config.ts (hunk 4 above)
Comment (kyscott18) [reply packages/create-ponder/tsup.config.ts:10]: Labels the information sent to the terminal

**Pull request merged**